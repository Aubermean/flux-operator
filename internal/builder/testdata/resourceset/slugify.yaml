apiVersion: fluxcd.controlplane.io/v1
kind: ResourceSet
metadata:
  name: app1
  namespace: apps
spec:
  inputs:
    - tenant: team1
      domain: t1.example.com
      semver: ">=1.0.0-rc.0"
      replicas: "2"
    - tenant: team2
      domain: t2.example.com
      semver: ">=1.0.0"
      replicas: "3"
  resources:
    - apiVersion: source.toolkit.fluxcd.io/v1beta2
      kind: OCIRepository
      metadata:
        name: << inputs.tenant >>-app1
        namespace: apps
        labels:
          app.kubernetes.io/semver: << inputs.semver | slugify >>
      spec:
        interval: 10m
        url: oci://ghcr.io/org/charts/app1
        ref:
          semver: << inputs.semver | quote >>
    - apiVersion: helm.toolkit.fluxcd.io/v2
      kind: HelmRelease
      metadata:
        name: << inputs.tenant >>-app1
        namespace: apps
        labels:
          replicas: << inputs.replicas | quote >>
      spec:
        interval: 1h
        releaseName: << inputs.tenant >>-app1
        chartRef:
          kind: OCIRepository
          name: << inputs.tenant >>-app1
        values:
          domain: << inputs.domain >>
          replicas: << inputs.replicas | int >>
