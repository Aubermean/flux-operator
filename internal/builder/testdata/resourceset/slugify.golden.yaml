apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  labels:
    app.kubernetes.io/semver: 1-0-0-rc-0
  name: team1-app1
  namespace: apps
spec:
  interval: 10m
  ref:
    semver: '>=1.0.0-rc.0'
  url: oci://ghcr.io/org/charts/app1
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  labels:
    app.kubernetes.io/semver: 1-0-0
  name: team2-app1
  namespace: apps
spec:
  interval: 10m
  ref:
    semver: '>=1.0.0'
  url: oci://ghcr.io/org/charts/app1
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  labels:
    replicas: "2"
  name: team1-app1
  namespace: apps
spec:
  chartRef:
    kind: OCIRepository
    name: team1-app1
  interval: 1h
  releaseName: team1-app1
  values:
    domain: t1.example.com
    replicas: 2
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  labels:
    replicas: "3"
  name: team2-app1
  namespace: apps
spec:
  chartRef:
    kind: OCIRepository
    name: team2-app1
  interval: 1h
  releaseName: team2-app1
  values:
    domain: t2.example.com
    replicas: 3
---
